services/github/github-downloads.service.js

```diff
@@ -24,173 +28,84 @@ const releaseArraySchema = Joi.alternatives().try(
   Joi.array().length(0),
 )
 
+const variantParam = pathParam({
```

Comment:
I've been kind of looking forward to getting this one sorted and simultaneously dreading the process of unpicking it.

Because we've got lots of different variants with slightly different combinations of the same params (all of which occur at least twice), I decided to pull them all out into variables and then lego them together.